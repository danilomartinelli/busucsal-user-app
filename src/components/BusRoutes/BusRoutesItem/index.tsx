import * as React from "react";
import { List, Divider } from "react-native-paper";
import { Route } from "../../../types/Route";

export interface IBusRoutesItemProps {
  routeHour: string;
  route: string;
  onPress: () => void;
}

const BusRoutesItem = ({ route, routeHour, onPress }: IBusRoutesItemProps) => (
  <React.Fragment>
    <List.Item
      title={route}
      description={routeHour}
      onPress={onPress}
      left={props => <List.Icon {...props} icon="access-time" />}
      right={props => <List.Icon {...props} icon="navigate-next" />}
    />
    <Divider />
  </React.Fragment>
);

export default BusRoutesItem;
