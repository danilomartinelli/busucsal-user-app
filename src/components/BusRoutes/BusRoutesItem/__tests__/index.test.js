import React from "react";
import { shallow } from "enzyme";
import BusRoutesItem from "..";

describe("<BusRoutesItem />", () => {
  const getProps = () => ({
    onPress: () => "",
    route: "essamsm",
    routeHour: "alguma",
    points: [],
  });

  describe("render", () => {
    it("should render properly", () => {
      const propsMock = getProps();

      const wrapper = shallow(<BusRoutesItem {...propsMock} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
