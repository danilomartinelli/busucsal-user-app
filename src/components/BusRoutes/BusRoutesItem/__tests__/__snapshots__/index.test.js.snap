// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusRoutesItem /> render should render properly 1`] = `ShallowWrapper {}`;
